src/server/graphql/mutations/inactivateUser.js

```diff
@@ -0,0 +1,79 @@
+import {GraphQLBoolean, GraphQLID, GraphQLNonNull} from 'graphql';
+import adjustUserCount from 'server/billing/helpers/adjustUserCount';
+import getRethink from 'server/database/rethinkDriver';
+import {requireOrgLeaderOfUser, requireWebsocket} from 'server/utils/authorization';
+import {toEpochSeconds} from 'server/utils/epochTime';
+import {MAX_MONTHLY_PAUSES, PAUSE_USER} from 'server/utils/serverConstants';
+import {errorObj} from 'server/utils/utils';
+
+export default {
+  type: GraphQLBoolean,
+  description: 'pauses the subscription for a single user',
+  args: {
+    userId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'the user to pause'
+    }
+  },
+  async resolve(source, {userId}, {authToken, socket}) {
+    const r = getRethink();
+
+    // AUTH
+    requireWebsocket(socket);
+    await requireOrgLeaderOfUser(authToken, userId);
+    const orgDocs = await r.table('Organization')
+      .getAll(userId, {index: 'orgUsers'})
+      .pluck('id', 'orgUsers', 'periodStart', 'periodEnd', 'stripeSubscriptionId');
+    const firstOrgUser = orgDocs[0].orgUsers.find((orgUser) => orgUser.id === userId);
+    if (!firstOrgUser) {
+      // no userOrgs means there were no changes, which means inactive was already true
+      throw errorObj({_error: 'That user is already inactive. cannot inactivate twice'});
+    }
+    const hookPromises = orgDocs.map((orgDoc) => {
+      const {periodStart, periodEnd, stripeSubscriptionId} = orgDoc;
+      const periodStartInSeconds = toEpochSeconds(periodStart);
+      const periodEndInSeconds = toEpochSeconds(periodEnd);
+      return r.table('InvoiceItemHook')
+        .between(periodStartInSeconds, periodEndInSeconds, {index: 'prorationDate'})
+        .filter({
+          stripeSubscriptionId,
+          type: PAUSE_USER,
+          userId
+        })
+        .count()
+        .run();
+    });
+    const pausesByOrg = await Promise.all(hookPromises);
+    const triggeredPauses = Math.max(...pausesByOrg);
+    if (triggeredPauses >= MAX_MONTHLY_PAUSES) {
+      throw errorObj({_error: 'Max monthly pauses exceeded for this user'});
```

Comment:
it's still a 200 I _think_. They are authorized to hit the GraphQL endpoint, everyone is. However, what happens inside that GraphQL query is a black box that returns `{data, errors}`.Since it's returning a useful value, although a value that they don't want, it's still 200. If we wanted to customize it, we could peek inside the graphql payload & adjust the response payload, but i kinda like the idea of using codes at the network layer-- did it connet to the server? did the server allow them to connect to graphql? OK, 200. The whole question boils down to: "If I successfully return an error, is it an error, or a success?".

Reply:
I see. I gotta familiarize myself with GraphQL's error reporting patterns.